synapse/federation/federation_client.py

```diff
@@ -1691,9 +1691,19 @@ async def _timestamp_to_event_from_destination(
                 #   to return events on *both* sides of the timestamp to
                 #   help reconcile the gap faster.
                 _timestamp_to_event_from_destination,
+                # Since this endpoint is new, we should try other servers before giving up.
+                # We can safely remove this in a year (remove after 2023-11-16).
+                failover_on_unknown_endpoint=True,
```

Comment:
Can we file an issue for this and add it to the Revisit: Next Year milestone?

Reply:
FWIW in the past I've just never removed the failover logic for new endpoints.

Reply:
-> https://github.com/matrix-org/synapse/issues/14577